tests/integration/api/test_geti_interaction.py

```diff
@@ -0,0 +1,209 @@
+        self.tmp_path = tmp_path
+        self.geti_config_path = geti_config_path
+        self.arrow_file_path = arrow_file_path
+        self.otx_config = self._convert_config()
+        self.engine, self.train_kwargs = self._instantiate_engine()
+        self.mapi_model_class = mapi_model_class
+        self.image = cv2.imread(str(image_path))
+
+    def _convert_config(self) -> dict:
+        otx_config = ConfigConverter.convert(config_path=self.geti_config_path)
+        otx_config["data"]["data_format"] = "arrow"
+        otx_config["data"]["train_subset"]["subset_name"] = "TRAINING"
+        otx_config["data"]["val_subset"]["subset_name"] = "VALIDATION"
+        otx_config["data"]["test_subset"]["subset_name"] = "TESTING"
+        return otx_config
+
+    def _instantiate_engine(self) -> tuple[Engine, dict[str, Any]]:
+        return ConfigConverter.instantiate(
+            config=self.otx_config,
+            work_dir=self.tmp_path,
+            data_root=self.arrow_file_path,
+        )
+
+    def test_model_and_data_module(self):
+        """Test the instance type of the model and the datamodule."""
+        assert isinstance(self.engine.model, OTXModel)
+        assert isinstance(self.engine.datamodule, OTXDataModule)
+
+    def test_training(self):
+        """Test the training process."""
+        max_epochs = 2
+        self.train_kwargs["max_epochs"] = max_epochs
+        train_metric = self.engine.train(**self.train_kwargs)
+        assert len(train_metric) > 0
+        assert self.engine.checkpoint
+
+    def test_predictions(self):
+        """Test the prediction process. This is way to check that the model is valid."""
+        predictions = self.engine.predict()
+        assert predictions is not None
+        assert len(predictions) > 0
+
+    def test_export_and_infer_onnx(self):
+        """Test exporting the model to ONNX."""
+        for precision in [OTXPrecisionType.FP16, OTXPrecisionType.FP32]:
+            exported_path = self.engine.export(
+                export_format=OTXExportFormatType.ONNX,
+                export_precision=precision,
+                explain=(precision == OTXPrecisionType.FP32),
+                export_demo_package=False,
+            )
+            export_dir = exported_path.parent
+            assert export_dir.exists()
+
+            # Test how model is loaded by Model API
+            if self.mapi_model_class is not None:
+                onnx_path = export_dir / "exported_model.onnx"
+                mapi_model = self.mapi_model_class.create_model(onnx_path)
+                assert mapi_model is not None
+
+                predictions = mapi_model(self.image)
+                assert predictions is not None
+                assert len(predictions.top_labels) > 0
+
+            exported_path.unlink(missing_ok=True)
+
+    def test_export_and_infer_openvino(self):
+        """Test exporting the model to OpenVINO."""
+        for precision in [OTXPrecisionType.FP16, OTXPrecisionType.FP32]:
```

Comment:
Can be done via pytest parametrization for verbosity

Reply:
@negvet @sovrasov Similar question. I think Geti support ONNX export? Do we need to test it as well?

Reply:
in this case I still need to manually loop through precision types in the `test_engine_api`.
The goal is also to do one training per both types check.